framework/src/systems/NonlinearEigenSystem.C

```diff
@@ -239,6 +239,13 @@ NonlinearEigenSystem::solve()
   if (_eigen_problem.isGeneralizedEigenvalueProblem() && sys().matrix_B)
     sys().matrix_B->close();
 #endif
+
+  if (_eigen_problem.needInitializeEigenVector())
+  {
+    _eigen_problem.initEigenvector(1.0);
+    _transient_sys.set_initial_space(solution());
```

Comment:
If we want to have the behavior in the old executioner, we should call this line always. and move the above line in `Eigenvalue::init`.

Reply:
It is always called. `if` is to check that it is indeed a nonlinear solver. We care about nonlinear solver at this point.